android/engine/src/main/java/org/smartregister/fhircore/engine/sync/SyncBroadcaster.kt

```diff
@@ -103,12 +103,26 @@ constructor(
   private fun Flow<PeriodicSyncJobStatus>.handlePeriodicSyncJobStatus(
     coroutineScope: CoroutineScope,
   ) {
-    this.onEach {
+    this.onEach { status ->
         syncListenerManager.onSyncListeners.forEach { onSyncListener ->
+          Timber.d("fhir sync worker...")
           onSyncListener.onSync(
-            if (it.lastSyncJobStatus != null) {
-              CurrentSyncJobStatus.Succeeded((it.lastSyncJobStatus as LastSyncJobStatus).timestamp)
-            } else it.currentSyncJobStatus,
+            if (status.lastSyncJobStatus != null) {
+              CurrentSyncJobStatus.Succeeded(
+                (status.lastSyncJobStatus as LastSyncJobStatus).timestamp,
+              )
+            } else status.currentSyncJobStatus,
+          )
+        }
+
+        if (status.currentSyncJobStatus is CurrentSyncJobStatus.Succeeded) {
```

Comment:
I think we should add an **OR** condition here with `status.lastSyncJobStatus != null` as well since it represents a `CurrentSyncJobStatus.Succeeded` status as well _(hence code block L110-L113)_

Reply:
Essentially we should be looking for two things, that is is not null and it Succeeded _(LastSyncJobStatus.Succeeded)_

Check out the Sealed class here - https://github.com/google/android-fhir/blob/507e89cc8b75c7b6667c8bff47b914c3ac94a770/engine/src/main/java/com/google/android/fhir/sync/SyncJobStatus.kt#L46